docs/design/simplified_federation_deployment.md

```diff
@@ -0,0 +1,388 @@
+
+Kubernetes Cluster Federation operates only on the clusters that it is
+responsible for. So once a federation control plane is deployed, for it
+to be useful, users should be able to register and deregister Kubernetes
+clusters with/from their federation. In order to register a cluster,
+users should supply the cluster's credentials and its API server
+endpoint.
+
+**Side note**: It is a mouthful to always prefix "federation" to each of
+its control plane components. But it is necessary to disambiguate these
+components from the almost identical components in a Kubernetes cluster.
+
+## Goals
+
+Today, we ship bash scripts to facilitate the federation control plane
+deployment process and the registration/deregistration process is mostly
+manual. Both these processes involve multiple coordinated steps and are
+error prone. For example, to register a Kubernetes cluster with
+federation users have to write a yaml file describing a cluster
+resource. They have to specify the cluster's endpoint in the right
+format. That field is too restrictive and hard to get it right. They
+also have to correctly specify where the credentials for the cluster is
+stored.
+
+Therefore, the goal here is to simplify both the federation control
+plane deployment experience and the cluster registration and
+deregistration process. Each of these processes will be reduced to a
+single command with additional flags for customization.
+
+
+## User Experience
+
+Kubernetes v1.4 shipped a new command line tool called
+[`kubeadm`](http://kubernetes.io/docs/getting-started-guides/kubeadm/).
+`kubeadm` simplifies Kubernetes control plane deployment. Federation
+control plane deployment builds on top of this tool. Cluster
+registration and deregistration will be facilitated through
+[`kubectl`](http://kubernetes.io/docs/user-guide/kubectl-overview/).
+
+
+#### Federation Control Plane Deployment Experience
+
+1. User provisions their cloud/compute resources: a VM or a physical
+   machine.
+2. Downloads the
+   [`kubeadm`](http://kubernetes.io/docs/getting-started-guides/kubeadm/)
+   executable on to each of those machines, virtual or physical.
+3. Runs:
+
+    ```shell
+    $ kubeadm init federation
+    ```
+
+   to set up a Kubernetes control plane and a federation control plane
+   on that machine.
+
+
+#### Cluster Registration Experience
+
+1. User downloads the
+   [`kubectl`](http://kubernetes.io/docs/getting-started-guides/kubeadm/)
+   executable on their local computer, i.e. the computer they use for
+   their day-to-day activities.
+2. Runs:
+
+    ```shell
+    $ kubectl register mycluster
+    ```
+
+   to register their cluster named `mycluster`, where `mycluster` is the
+   name of the cluster's context in the local kubeconfig.
+
+#### Cluster Deregistration Experience
+
+1. To deregister a cluster, user runs:
+
+    ```shell
+    $ kubectl deregister mycluster
+    ```
+
+### Points for Discussion:
+
+1. Is `unregister` a better verb than `deregister`?
```

Comment:
Personally I prefer "deregister", but both appear to be in common use, and equally correct language.